comp/core/workloadmeta/collectors/internal/process/process_collector_test.go

```diff
@@ -0,0 +1,280 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux && test
+
+// Package process implements the local process collector for
+// Workloadmeta.
+package process
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/comp/core"
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	workloadmetafxmock "github.com/DataDog/datadog-agent/comp/core/workloadmeta/fx-mock"
+	workloadmetamock "github.com/DataDog/datadog-agent/comp/core/workloadmeta/mock"
+	"github.com/DataDog/datadog-agent/pkg/languagedetection/languagemodels"
+	processwlm "github.com/DataDog/datadog-agent/pkg/process/metadata/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil/mocks"
+	proccontainers "github.com/DataDog/datadog-agent/pkg/process/util/containers/mocks"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+)
+
+// NewProcessDataWithMockProbe returns a new ProcessData with a mock probe
+func NewProcessDataWithMockProbe(t *testing.T) (*Data, *mocks.Probe) {
+	probe := mocks.NewProbe(t)
+	return &Data{
+		probe: probe,
+	}, probe
+}
+
+type collectorTest struct {
+	collector *collector
+	probe     *mocks.Probe
+	mockClock *clock.Mock
+	mockStore workloadmetamock.Mock
+}
+
+func setUpCollectorTest(t *testing.T, configOverrides map[string]interface{}) collectorTest {
+	mockStore := fxutil.Test[workloadmetamock.Mock](t, fx.Options(
+		core.MockBundle(),
+		fx.Replace(config.MockParams{Overrides: configOverrides}),
+		fx.Supply(workloadmeta.Params{
+			AgentType: workloadmeta.NodeAgent,
+		}),
+		workloadmetafxmock.MockModule(),
+	))
+
+	time.Sleep(time.Second)
```

Comment:
Any reason for this? if so lets document with a comment. thanks!

Reply:
Taken from another WLM test, but I deleted it as it still worked.